Services/Database/classes/Setup/class.ilMysqlMyIsamToInnoDbMigration.php

```diff
@@ -79,11 +79,16 @@ public function step(Environment $environment): void
     {
         $rows = $this->getNonInnoDBTables();
         $table_name = array_pop($rows);
-        if (is_string($table_name) && strlen($table_name) > 0) {
-            $migration = $this->database->migrateTableToEngine($table_name);
-        }
-        if (isset($migration) && $migration === false) {
-            throw new ilException("The migration of the following tables did throw errors, please resolve the problem before you continue: \n" . $table_name);
+
+        if (is_string($table_name) && $table_name !== '') {
+            try {
+                $this->database->migrateTableToEngine($table_name);
+            } catch (\ilDatabaseException $e) {
+                throw new ilException(
```

Comment:
@mjansenDatabay Should we use a `UnachievableException` exception here? I can change this while merging it

Reply:
Good point @chfsx , a `UnachievableException` would be indeed more appropriate.